src/components/Swap/SwapButton/index.tsx

```diff
@@ -67,80 +67,82 @@ export default memo(function SwapButton({ disabled }: SwapButtonProps) {
   // Close the review modal on chain change.
   useEffect(() => setOpen(false), [chainId])
 
-  const addTransaction = useAddTransaction()
+  const addTransactionInfo = useAddTransaction()
   const setDisplayTxHash = useUpdateAtom(displayTxHashAtom)
-  const setOldestValidBlock = useSetOldestValidBlock()
+  const [, setSwapAmount] = useSwapAmount(Field.INPUT)
+  const resetSwapAmount = useCallback(() => setSwapAmount(''), [setSwapAmount])
+
+  // Submits a transaction. Returns true if the transaction was submitted.
+  const onSubmit = useCallback(
+    async (submit: () => Promise<TransactionInfo | undefined>): Promise<boolean> => {
+      let info: TransactionInfo | undefined
+      try {
+        info = await submit()
+      } catch (e) {
+        console.error('Failed to submit', e)
+      }
+      if (!info) return false
+
+      resetSwapAmount()
+      addTransactionInfo(info)
+      setDisplayTxHash(info.response.hash)
+
+      if (isAnimating(document)) {
+        // Only return after any queued animations to avoid layout thrashing, because a successful
+        // submit will open the status dialog and immediately cover the animating button.
+        return new Promise((resolve) => {
+          const onAnimationEnd = () => {
+            document.removeEventListener('animationend', onAnimationEnd)
+            resolve(true)
+          }
+          document.addEventListener('animationend', onAnimationEnd)
+        })
+      }
+      return true
+    },
+    [addTransactionInfo, resetSwapAmount, setDisplayTxHash]
+  )
 
   const [isPending, setIsPending] = useState(false)
   const native = useNativeCurrency()
   const onWrap = useCallback(async () => {
     setIsPending(true)
-    try {
-      const transaction = await wrapCallback()
-      if (!transaction) return
-      invariant(wrapType !== undefined)
-      addTransaction({
-        response: transaction,
-        type: wrapType,
-        amount: CurrencyAmount.fromRawAmount(native, transaction.value?.toString() ?? '0'),
-      })
-      setDisplayTxHash(transaction.hash)
-      // Only reset pending after any queued animations to avoid layout thrashing, because a
-      // successful wrap will open the status dialog and immediately cover the button.
-      const postWrap = () => {
-        setIsPending(false)
-        document.removeEventListener('animationend', postWrap)
-      }
-      if (isAnimating(document)) {
-        document.addEventListener('animationend', postWrap)
-      } else {
-        setIsPending(false)
-      }
-    } catch (e) {
-      // TODO(zzmp): Surface errors from wrap.
-      console.log(e)
-    } finally {
-      setIsPending(false)
-    }
-  }, [addTransaction, native, setDisplayTxHash, wrapCallback, wrapType])
+    await onSubmit(async () => {
+      const response = await wrapCallback()
+      if (!response) return
+
+      invariant(wrapType)
```

Comment:
This is a subtly different assertion than the one being removed. `WrapType.NONE` (`0`) will throw here now. Is that intentional?

Reply:
`WrapType.NONE` was removed in https://github.com/Uniswap/widgets/pull/159 - this is now a type alias for `TransactionType.WRAP | TransaactionType.UNWRAP | undefined`.

I can (and should) still make this more explicit, so I'll update to `wrapType !== undefined`.